msdsn/conn_str_test.go

```diff
@@ -196,3 +200,68 @@ func TestConnParseRoundTripFixed(t *testing.T) {
 		t.Fatal("Parameters do not match after roundtrip", params, rtParams)
 	}
 }
+
+func TestInvalidConnectionStringKerberos(t *testing.T) {
+	connStrings := []string{
+		"server=server;port=1345;realm=domain;trustservercertificate=true;keytabfile=/path/to/administrator2.keytab;enablekerberos=true",
+		"server=server;port=1345;realm=domain;trustservercertificate=true;krbcache=;enablekerberos=true",
+		"server=server;user id=user;password=pwd;port=1345;realm=domain;trustservercertificate=true;krb5conffile=/etc/krb5.conf;enablekerberos=true",
+		"server=server;user id=user;password=pwd;port=1345;realm=domain;trustservercertificate=true;krb5conffile=/etc/krb5.conf;keytabfile=/path/to/administrator2.keytab;enablekerberos=true",
+		"server=server;user id=user;port=1345;realm=domain;trustservercertificate=true;krb5conffile=/etc/krb5.conf;keytabfile=/path/to/administrator2.keytab;enablekerberos=true;initkrbwithkeytab=false",
+		"server=server;user id=user;port=1345;realm=domain;trustservercertificate=true;krb5conffile=/etc/krb5.conf;enablekerberos=true;initkrbwithkeytab=true",
+	}
+	for _, connStr := range connStrings {
+		_, _, err := Parse(connStr)
+		if err == nil {
+			t.Errorf("Connection expected to fail for connection string %s but it didn't", connStr)
+			continue
+		} else {
+			t.Logf("Connection failed for %s as expected with error %v", connStr, err)
+		}
+	}
+}
+
+func TestValidConnectionStringKerberos(t *testing.T) {
```

Comment:
This seems like a test that doesn't actually test for much. Also, if all the files are going to be bogus files, just have them point to the same file. Also seems like any temp files should be written to the temp folder.